pkg/resources/custom-smtp/custom-smtp.go

```diff
@@ -0,0 +1,233 @@
+package custom_smtp
+
+import (
+	"context"
+	"fmt"
+	"github.com/integr8ly/integreatly-operator/apis/v1alpha1"
+	"github.com/integr8ly/integreatly-operator/pkg/addon"
+	corev1 "k8s.io/api/core/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	Valid        ValidationResponse = "valid"
+	Partial      ValidationResponse = "partial"
+	Blank        ValidationResponse = "blank"
+	CustomSecret string             = "custom-smtp"
+)
+
+type ValidationResponse string
+
+type CustomSmtp struct {
+	FromAddress       string
+	Address           string
+	Password          string
+	Port              string
+	OpensslVerifyMode string
+	Username          string
+}
+
+func GetCustomAddonValues(serverClient k8sclient.Client, namespace string) (*CustomSmtp, error) {
+
+	addonName := addon.GetName(v1alpha1.InstallationTypeManagedApi)
+	secretName := fmt.Sprintf("addon-%s-parameters", addonName)
+
+	secret := &corev1.Secret{}
```

Comment:
Could we add a function for getting the secret. It's used a few times in this file.

Reply:
Yes, it's a code neatness item so wouldn't block the change for it.

Reply:
I am going to be creating a few follow up tickets. This can be one and I think it could be a good first issue also.